hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -192,6 +192,10 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
         throw new OMException("Bucket already exist", BUCKET_ALREADY_EXISTS);
       }
 
+      //Check quotaInBytes and quotaInCounts to update
+      checkQuotaBytesValid(omVolumeArgs, omBucketInfo);
+      checkQuotaCountsValid(omVolumeArgs, omBucketInfo);
```

Comment:
Argument validity should be checked before acquiring lock, preferably in `preExecute`.

Also, please verify that the bucket is not a link, if quota is set in the request.  Links cannot have actual content, so they should not have any quota defined, similar to encryption:

https://github.com/apache/hadoop-ozone/blob/079ee7fc2a223e1251b16b9c42004aa2a27bf0f4/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java#L127-L130

Reply:
Thanks @adoroszlai for the advice, the check action needs to get the volume or bucket in DB, so it better to do this action after acquiring the lock.  And other comments has been fixed.